clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubProtocol.java

```diff
@@ -16,7 +18,7 @@
      * @param message A string representation of one or more {@link HubMessage}s.
      * @return A list of {@link HubMessage}s.
      */
-    HubMessage[] parseMessages(String message);
+    HubMessage[] parseMessages(String message, InvocationBinder binder) throws Exception;
```

Comment:
Checked exceptions...

![Eye roll](https://media.giphy.com/media/3h5pe45FM9qUM/giphy.gif)

Reply:
I think we should have a wrapper exception type here, but that can wait for The Great Exception Reckoning (#2610)